File: 02-backend/src/main/java/com/springbootshopping/app/dto/Purchase.java
package com.springbootshopping.app.dto;

import java.util.Set;

import com.springbootshopping.app.entity.Address;
import com.springbootshopping.app.entity.Customer;
import com.springbootshopping.app.entity.Order;
import com.springbootshopping.app.entity.OrderItem;

import lombok.Data;

@Data
public class Purchase {

    private Customer customer;
    private Address shippingAddress;
    private Address billingAddress;
    private Order order;
    private Set<OrderItem> orderItems;

}

File: 03-frontend/angular-shoppingapp/src/app/common/purchase.ts
import { Address } from './address';
import { Customer } from './customer';
import { Order } from './order';
import { OrderItem } from './order-item';

export class Purchase {
    customer: Customer | undefined;
    shippingAddress: Address | undefined;
    billingAddress: Address | undefined;
    order: Order | undefined;
    orderItems: OrderItem[] | undefined; 
}
